### PR TITLE
Fixes: #4957 --- Intellisense for Indexed properties were broken by a recent 

### PR DESCRIPTION
**Pull request by KevinRansom**
image

### CODE REVIEW

File: src/fsharp/infos.fs [hunk 1]
```diff
@@ -1005,7 +1003,8 @@ type MethInfo =
     member x.FormalMethodTypars = 
         match x with 
         | ILMeth(_,ilmeth,_) -> ilmeth.FormalMethodTypars
-        | FSMeth(g,typ,vref,_) ->  
+        | FSMeth(g,_,vref,_) ->  
+            let typ = x.ApparentEnclosingAppType
```
Comment (dsyme) [inline src/fsharp/infos.fs]: Ugh, yes. However there are a few other paths to `AnalyzeTypeOfMemberVal`, which is assuming that `typ` is an AppTy, e.g. via `GetCompiledReturnTy` and `GetParamTypes` which I think will also fail.  I'm surprised we're not hitting those

File: src/fsharp/infos.fs [hunk 2]
```diff
@@ -897,9 +897,7 @@ type MethInfo =
 
     /// Get the enclosing type of the method info, using a nominal type for tuple types
     member x.ApparentEnclosingAppType = 
-        match x with
-        | ILMeth(_,ilminfo,_) -> ilminfo.ApparentEnclosingAppType
-        | _ -> x.ApparentEnclosingType
+        helpEnsureTypeHasMetadata x.TcGlobals x.ApparentEnclosingType
```
[the remark this answers is not in the document]
Comment (dsyme) [reply src/fsharp/infos.fs]: @forki It doesn't guarantee to convert to a type with metadata, but for function and tuple types it does.  So it's something like `convertToTypeWithMetadataIfPossible`.

File: src/fsharp/infos.fs (hunk 1 above)
Comment (KevinRansom) [reply src/fsharp/infos.fs]: @dsyme, yes I was going to look through those today.  The trick is figuring out the repro, as you say, with all this code base none hit it.

**Comment by KevinRansom** [comment]
@dsyme, can you take a look at the update changes, I think I have the right paths covered now.  Although, I couldn't write any failing repros.

**Comment by dsyme** [comment]
That looks good.

**Pull request merged**